File: infrastructure/src/main/java/org/haas/infrastructure/persistence/seat/repository/SeatRepositoryImpl.java
package org.haas.infrastructure.persistence.seat.repository;

import lombok.RequiredArgsConstructor;
import org.haas.core.domain.seat.Seat;
import org.haas.core.domain.seat.repository.SeatRepository;
import org.haas.infrastructure.persistence.seat.entity.SeatEntity;
import org.springframework.stereotype.Repository;

@RequiredArgsConstructor
@Repository
public class SeatRepositoryImpl implements SeatRepository {

    private final SeatJpaRepository seatJpaRepository;

    @Override
    public Seat findByRowAndColumn(int row, int column) {
//        SeatEntity seatEntity = seatJpaRepository.findByRowAndColumn(row, column)
//                .orElse(null);
//        if (seatEntity != null) {
//            return seatMapper.toDomain(seatEntity);
//        }
        return null;
    }
}

File: movie/infra/src/main/kotlin/com/example/movie/persistence/seat/repository/SeatRepositoryImpl.kt
package com.example.movie.persistence.seat.repository

import com.example.movie.domain.seat.model.Seat
import com.example.movie.domain.seat.repository.SeatRepository
import org.springframework.stereotype.Repository

@Repository
class SeatRepositoryImpl(
    private val seatJpaRepository: SeatJpaRepository
) : SeatRepository {
    override fun findAllByIdIn(seatIds: Collection<Long>): List<Seat> {
        val seatEntities = seatJpaRepository.findAllByIdIn(seatIds)
        return seatEntities.map{ it.toDomain()}
    }
}